projects/plugins/jetpack/tools/build-block-manifest.php

```diff
@@ -0,0 +1,91 @@
+<?php // phpcs:ignore Squiz.Commenting.FileComment.MissingPackageTag
+/**
+ * Build script to generate a PHP manifest file containing block metadata from block.json files.
+ * This reduces filesystem reads and JSON parsing at runtime.
+ */
+
+// phpcs:disable WordPress.WP.AlternativeFunctions, WordPress.PHP.DevelopmentFunctions.error_log_var_export
+
+/**
+ * Generates a manifest file containing block metadata from block.json files.
+ *
+ * Scans the given directory for block.json files, combines their metadata
+ * into a single PHP file that returns an array. This improves performance
+ * by avoiding filesystem reads and JSON parsing at runtime.
+ *
+ * @param string $base_path Optional. Base directory to scan for block.json files.
+ *                         Defaults to Jetpack blocks directory.
+ * @return bool True on success, throws Exception with error message on failure.
+ * @throws Exception If directory doesn't exist, no files found, or write fails.
+ */
+function build_block_manifest( $base_path = null ) {
+	if ( null === $base_path ) {
+		$base_path = __DIR__ . '/../extensions/blocks';
+	}
+
+	if ( ! file_exists( $base_path ) ) {
+		throw new Exception( "Input directory does not exist: {$base_path}", 1 );
+	}
+
+	$blocks = array();
+	$files  = glob( $base_path . '/**/block.json' );
```

Comment:
I don't think PHP's `glob` understands `**` (other than as two `*`s), and if it did I think it would be wrong anyway because it seems WP only allows one level of directories.
```suggestion
	$files  = glob( $base_path . '/*/block.json' );
```